src/averageAges.js

```diff
@@ -37,7 +47,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenAgeSum = people
+    .filter(person =>
+      withChildren
+        ? people.some(child => child.mother === person.name)
+        && person.sex === 'f'
+        : person.sex === 'f'
+    );
+
+  return womenAgeSum
+    .reduce((sum, person) =>
+      sum + (person.died - person.born), 0)
```

Comment:
GJ, approving. One small suggestion, maybe will be better to make one more function to calculate average age in one place and reuse it all cases?)

Reply:
Good tip, u use the same condition in **calculateMenAverageAge**
![image](https://github.com/mate-academy/js_average-ages/assets/54936044/5f4df6a4-b695-41db-8161-0454781cf899)